lib/simplehttp.ex

```diff
@@ -140,19 +125,38 @@ defmodule SimpleHttp do
   end
 
   defp add_http_options_to_request(%Request{} = request, args) do
-    keys = [:timeout, :connect_timeout, :autoredirect]
-
-    http_options =
-      Enum.reduce(keys, [], fn key, acc ->
-        case args[key] do
-          nil -> acc
-          value -> acc ++ [{key, value}]
-        end
-      end)
+    keys = [
+      :timeout, :connect_timeout, :autoredirect,
+      :ssl, :essl, :proxy_auth, :version, :relaxed
+    ]
+    http_options = filter_options(keys, args)
 
     %{request | http_options: http_options}
   end
 
+  defp add_options_to_request(%Request{} = request, args) do
+    keys = [
+      :sync, :stream, :body_format, :full_result, :headers_as_is,
+      :socket_opts, :receiver, :ipv6_host_with_brackets
+    ]
+    options = filter_options(keys, args)
+
+    %{request | options: options}
+  end
+
+  defp add_global_options(%Request{} = request, args) do
+    keys = [
+      :proxy,                 :https_proxy,           :max_sessions,
```

Comment:
wondering if you will run `mix format` if this code block will look the same

Reply:
Since the project wasn't set up with the `.formatter.exs`, I didn't think you wanted to support mix formatting by default, as currently we get this:
```
$ mix format
** (Mix) Expected one or more files/patterns to be given to mix format or for a .formatter.exs file to exist with an :inputs or :subdirectories key
```

Reply:
when I started the project formatter was not included :)